pkgs/tools/package-management/nix/default.nix

```
@@ -234,13 +234,13 @@ in rec {
   nixUnstable = lib.lowPrio (callPackage common rec {
     pname = "nix";
     version = "2.4${suffix}";
-    suffix = "pre20210802_47e96bb";
+    suffix = "pre20210902_77ca5e9";
```

```suggestion
    suffix = "pre20210902_${lib.substring 0 7 src.rev}";
```
see https://github.com/NixOS/nixpkgs/pull/136271/files#diff-b94b72b12e07bed8c4a3b4c50242499d5efb1e32797426d319574fd613bc07aeR243